server/utils/world/getWorldDetails.ts

```diff
@@ -15,10 +15,11 @@ export const getWorldDetails = async (credentials: Credentials) => {
     const sceneDropId = credentials.sceneDropId || assetId
 
     const world = World.create(urlSlug, { credentials });
-    await world.fetchDetails();
-    await world.fetchDataObject();
 
-    await initializeWorldDataObject({ credentials, world });
+    await Promise.all([
```

Comment:
I saw you moved world.fetchDataObject() inside the initializeWorldDataObject.js file. This works, But I think that the right place to fetch the data object is here, and not inside the initializeWorldDataObject

Like
`await Promise.all(world.fetchDetails(), world.fetchDataObject())`;